src/components/nav-icon-item/nav-icon-item.tsx

```diff
@@ -0,0 +1,63 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, FC, MouseEventHandler, ReactNode } from 'react'
+import { ElAnchorNavIconItem, ElButtonNavIconItem } from './styles'
+
+interface CommonNavIconItemProps {
+  /**
+   * The JSX element to render as the icon
+   */
+  icon: ReactNode
+
+  /**
+   * Whether the nav item is active
+   * @default false
+   **/
+  isActive?: boolean
+}
+
+interface NavIconItemAsButtonElementProps extends CommonNavIconItemProps, ButtonHTMLAttributes<HTMLButtonElement> {
+  href?: never
+  target?: never
+  ref?: never
```

Comment:
**question:** why is `ref` being prevented?

Reply:
my bad, I misspelled it, it should be `rel` instead of `ref`

`rel` only valid on `<form>`,  `<link>`, `<area>`, and `<form>`; so I would exclude them for a `button`